xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
> @balamurugana, not sure how this method makes difference to avoid stale locks. Anyway, if you want to avoid deadlocks, you can drop mutexes and use actor model instead which should be simple with go

The stale locks are differentiated by keeping them in a map and purging them subsequently based on the active count and understanding if they are InUse.  

An actor model involves using channels - which is fine as well but sync primitives are better suited here.